pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -43,12 +45,45 @@ type nodeConfig struct {
 }
 
 // NewNodeConfig creates a new instance of nodeConfig to be used by the caller.
-func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM) *nodeConfig {
-	return &nodeConfig{k8sclientset: clientset, Windows: windows.New(windowsVM)}
+func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM) (*nodeConfig, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get config to talk to kubernetes api server")
+	}
+	kubeAPIServerEndpoint := cfg.Host
+	if len(kubeAPIServerEndpoint) == 0 {
+		return nil, errors.Wrap(err, "could not get host name for the kubernetes api server")
+	}
+	clusterAddress, err := getClusterAddr(kubeAPIServerEndpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting cluster address")
+	}
```

Comment:
All this (lines 49-60) belongs inside `getClusterAddr()`. There is no reason to pass `kubeAPIServerEndpoint` into `getClusterAddr()` as there is no other use for the intermediate variables in this scope. Or was this done to make it unit testable? In that case please add this code to a helper `getKuberAPIServerEndpoint()`.

I am also not sure if we want to do this for every node that is being added. Shouldn't discovering the cluster address be a one time boot up operation? Or is it possible to change the cluster address without tearing it down i.e. during the lifetime of a cluster?